core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1674,15 +1674,40 @@ protected void unparseDateTime(SqlWriter writer, SqlCall call, int leftPrec, int
         .getValue() : call.operand(0).toString();
     SqlOperator function = call.getOperator();
     if (!dateFormat.contains("%")) {
+      SqlNode modifiedSecondOperand = modifySecondOperand(call.operand(1));
       SqlCall formatCall =
           function.createCall(SqlParserPos.ZERO,
-                  createDateTimeFormatSqlCharLiteral(dateFormat), call.operand(1));
+              createDateTimeFormatSqlCharLiteral(dateFormat), modifiedSecondOperand);
       function.unparse(writer, formatCall, leftPrec, rightPrec);
     } else {
       function.unparse(writer, call, leftPrec, rightPrec);
     }
   }
 
+  private SqlNode modifySecondOperand(SqlNode sqlNode) {
+    if (sqlNode instanceof SqlCall && isConcatOperator((SqlCall) sqlNode)) {
+      List<SqlNode> modifiedOperands = getModifiedOperands((SqlCall) sqlNode);
+      return SqlStdOperatorTable.CONCAT.createCall(SqlParserPos.ZERO, modifiedOperands);
+    }
+    return sqlNode;
+  }
+
+  private static boolean isConcatOperator(SqlCall sqlNode) {
+    return sqlNode.getOperator().getKind() == SqlKind.CONCAT;
+  }
+
+  private List<SqlNode> getModifiedOperands(SqlCall call) {
```

Comment:
modify to stream if possible 

 return call.getOperandList().stream()
            .map(operand -> operand.getKind() == SqlKind.FORMAT ? ((SqlBasicCall) operand).getOperandList().get(1) : operand)
            .collect(Collectors.toList());